core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -119,7 +124,24 @@ public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContex
         return generatedContexts.stream();
     }
 
+    @Override
+    public void beforeEach(ExtensionContext context) {
+        detectThreadLeak = DetectThreadLeak.of(thread -> {
+            String name = thread.getName();
+            return !name.startsWith("metrics-meter-tick-thread") &&
+                    !name.startsWith("scala-") &&
+                    !name.startsWith("ForkJoinPool") &&
+                    !name.startsWith("junit-") &&
+                    !name.startsWith("Attach Listener") &&
+                    !name.startsWith("process reaper");
+        });
+    }
 
+    @Override
+    public void afterEach(ExtensionContext context) throws InterruptedException {
+        TestUtils.waitForCondition(() -> detectThreadLeak.newThreads().isEmpty(),
+                "Thread leak detected: " + detectThreadLeak.newThreads());
```

Comment:
maybe we can show the thread name in the log message?

Reply:
Yeah, `Thread#toString` already show `getName` in it. I change this line to join each thread name. Hope it's more clear. Thanks.